clients/pkg/logentry/stages/match.go

```diff
@@ -114,11 +114,23 @@ func newMatcherStage(logger log.Logger, jobName *string, config interface{}, reg
 }
 
 func getDropCountMetric(registerer prometheus.Registerer) *prometheus.CounterVec {
+	return getLogEntryMetric(registerer, "dropped_lines_total",
+		"A count of all log lines dropped as a result of a pipeline stage",
+		[]string{"reason"})
+}
+
+func getDropCountByLabelMetric(registerer prometheus.Registerer) *prometheus.CounterVec {
+	return getLogEntryMetric(registerer, "dropped_lines_by_label",
```

Comment:
In intentended usecase it will not be high cardinality metric. In case of pathalogical misconfiguration, like ratelimiting based on trace id, we reset all counts every  `max_distinct_labels` unique labels values processed to prevent OOM.

Reply:
I'd like to remind that ratelimiting itself is there to prevent excessive resource utilization in case of a misbehaving app. In a normal case no ratelimiting will happen and no dropped count metric will ever be created. Existence of the dropped counter by metric is a rare occasion and is definitely of interest to system administrators.

Dropped counter for every label when label itself is high cardinality is a rare occasion squared, but even that case is now handled.